event-sourcing/event-repository/event-repository-jdbc/src/main/java/uk/gov/justice/services/eventsourcing/repository/jdbc/JdbcEventRepository.java

```diff
@@ -90,9 +94,20 @@ public long getCurrentSequenceIdForStream(final UUID streamId) {
                     streamIds.onClose(eventStream::close);
                     return eventStream.map(eventConverter::envelopeOf);
                 });
+    }
 
+    @Override
+    public Stream<Stream<JsonEnvelope>> getStreamOfAllActiveEventStreams() {
+        final Stream<UUID> streamIds = eventStreamJdbcRepository.findActive().map(es -> es.getStreamId());
+        return streamIds
+                .map(id -> {
+                    final Stream<Event> eventStream = eventJdbcRepository.findByStreamIdOrderBySequenceIdAsc(id);
+                    streamIds.onClose(eventStream::close);
+                    return eventStream.map(eventConverter::envelopeOf);
+                });
```

Comment:
The code above is duplicated in another method in the class...could be a method

Reply:
Done